sklearn/metrics/_classification.py

```diff
@@ -2089,55 +2134,132 @@ class after being classified as negative. This is the case when the
             f"problems, got targets of type: {y_type}"
         )
 
+    if raise_warning != "deprecated":
+        warnings.warn(
+            "`raise_warning` was deprecated in version 1.7 and will be removed "
+            "in 1.9, when an `UndefinedMetricWarning` will always raise in case of a "
+            "division by zero.",
+            FutureWarning,
+        )
+    else:
+        if replace_undefined_by == "default":
+            # TODO(1.9): Remove. If users don't set any return values in case of a
+            # division by zero (`raise_warning="deprecated"` and
+            # `replace_undefined_by="default"`) they still get a FutureWarning about
+            # changing default return values:
+            warnings.warn(
+                "The default return value of `class_likelihood_ratios` in case of a "
+                "division by zero has been deprecated in 1.7 and will be changed to "
+                "the worst scores (`(1.0, 1.0)`) in version 1.9. Set "
+                "`replace_undefined_by='worst'` to use the new default and to silence "
+                "this Warning.",
+                FutureWarning,
+            )
```

Comment:
Hm, I am not sure. Right before we already raise a `FutureWarning` if the user has set `raise_warning`, but with another warning message. The problem is that the user who uses  `raise_warning` is not aware of it's default setting behavior (since there was no other choice than `np.nan` before). So I think it makes sense to have two different warning messages for both cases.

I have just added mentioning of `replace_undefined_by` in the first warning. What do you think?

Reply:
I'm okay with always raising only one warning, but that single warning should then tell them about both changes. The issue right now is that if they set `raise_warning`, they don't see any information about the default value returned in case of a division by zero.

Reply:
Yes, it's good to show this information as well. But maybe let's not bother users who did NOT use the deprecated param with a warning about it. This is why I changed the warning behaviour as follows:

if users use the old `raise_warning`:
`    `msg_deprecated_param+mgs_changed_default

if they don't set neither `raise_warning` nor `replace_undefined_by`:
`    `mgs_changed_default

What do you think, @adrinjalali?